book/3-collab-dev-platforms/collaborative_ml.md

```diff
@@ -0,0 +1,35 @@
+
+With the increasing complexity of Machine Learning projects, highly likely a team will be formed to complete the task. Being able to collaborate efficiently may be more critical than ever before. 
+
+Fundamentally, there are two ways that teams collaborate. <br>
+*Luke Marsden: When you boil it down to basics, there are really two fundamental modes of collaborating between different people doing work: synchronous and asynchronous.*
+
+*Synchronous*: Communications are scheduled, real-time interactions by phone, video, or in-person, sometimes interrupting one another for urgent tasks or simply asking a question. In terms of development, the upstream have to complete their tasks before the downstream team can continue with their work (Dependencies issues)
+
+*Asynchronous*: Communication happens on your own time, with a different team or people concurrently working on a different task. An efficient coping mechanism is required to resolve merged conflict. 
+
+With this in mind, we seek to introduce some tools and practices for a team to get started in the hope that stumbling blocks will be reduced to the minimum. The practices introduce here assume development using python as the primary language.
+
+## Communication tools.
+Establishing effective communication right from the beginning of any project will allow all team members to align toward the same goal and reduce any possible friction. An example of a quintessential miscommunication is an unclear problem statement leading to team members having different views on the outcome of the projects.
+The following tools by no means exhaustive proved to be useful for remote, office, and hybrid working arrangements.
+- A big whiteboard often produces the most efficient channel to share ideas. (Physical Meetup)
+- [ Google Workspace](https://workspace.google.com/), a full suite of communication tools such as Google meet, shared Google calendar, and Gmail to see the complete list of participants and scheduled meetings. Other option includes the [Microsoft Team](https://www.microsoft.com/en-us/microsoft-teams/log-in).
+- [Miro](https://miro.com/) is an easy to use, intuitive, collobration tool that uses little effort to setup quickly. Other option includes [Coggle](https://coggle.it/).
+
+## Code versioning
+Jupyter notebook(.ipynb) although powerful has many limitations. Versioning is not one of them and is certainly not the best format to be deployed or go into production. Name and renaming files or folders with comments are also undesirable and difficult to keep track of. Instead, using [git](https://git-scm.com/) for distributed version control and tracking changes helps coordinate work among team members. Developing code in python script (.py) is highly preferred and using [Gitlab](https://about.gitlab.com/) or [Github](https://github.com/) to manage your code online with an advanced interface to resolve merged conflict that occurs frequently for asynchronous work. 
```

Comment:
```Jupyter notebook(.ipynb) although powerful has many limitations. Versioning is not one of them...```
Do you mean to say versioning is one of them?
```, and it is certainly not the best format to be deployed or to go into production. Naming and renaming...```
Edited as above.

Reply:
Updated

Jupyter notebook(.ipynb) although powerful has many limitations. `For example, it is impractical to use pure git for versioning Jupyter notebook,` and it is certainly not the best format to be deployed or go into production. `Naming and renaming`